### DISCUSSION TITLE
Question about "Why SFTP is still slow in curl" post

### DISCUSSION THREAD

**Discussion by MarvinWinks** [curl/curl#9336] [General]
I've observed slowness with SFTP thru libcurl and came across https://daniel.haxx.se/blog/2014/05/14/why-sftp-is-still-slow-in-curl/. 

I was wondering if all of the contents of the posting are still relevant, given that the posting is over 12 years old. Most notably the comments about a possible solution.

Thanks!

**Comment by gvanem** [comment]
> I was wondering if all of the contents of the posting are still relevant, given that the posting is over 12 years old.

No. There was a long discussion about it [here](https://curl.se/mail/lib-2018-08/0019.html) 4 years ago. 

In this discussion, I made a monolithic sftp enabled `curl.exe` that's very fast. It's [here](  http://www.watt-32.net/misc/curl-monolitic.zip). 3.47 MBytes/sec upload (Norway to Denmark). 

It was a simple patch. Basically increasing `UPLOAD_BUFSIZE` to 512 kByte. Which I believe now is between 16 kByte and 2 MByte.
Ref: `UPLOADBUFFER_*` in `lib/urldata.h`.

**Reply by MarvinWinks** [reply]
Thanks for the link/info.

The reason for me posting this question was I was still observing slow SFTP download speeds with the current version of libcurl (and curl). @bagder's recommendation to increase CURL_MAX_WRITE_SIZE does improve performance (greatly), but as per his article, "that’s neither a nice nor clean fix". 

I was wondering if his recommendation still applies (is still the proper fix):

> A proper fix should first of all only be applied for SFTP transfers, thus deciding at run-time if it is necessary or not. Then it should dynamically provide a larger buffer and thirdly, for upload it should probably make the buffer “sliding” as in the libssh2 example code [sftp_write_sliding.c](http://libssh2.org/examples/sftp_write_sliding.html).

Thanks

**Reply by bagder** [reply]
These days we can set [CURLOPT_BUFFERSIZE](https://curl.se/libcurl/c/CURLOPT_BUFFERSIZE.html), which should help.

**Reply by MarvinWinks** [reply]
Thanks @bagder. Works great!
